File: Services/TaskService/TaskService.Application/DTOs/TaskDataDTO.cs
namespace TaskService.Application.DTOs
{
    public class TaskDataDTO
    {
        public int Id { get; set; }
        public int? CompanyId { get; set; }
        public int? ParentId { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public decimal? Budget { get; set; }
        public Domain.Enums.TaskStatus? Status { get; set; }
        public DateOnly? CompleteDate { get; set; }
        public UserDTO[]? Members { get; set; }

        public ICollection<ResourceDTO>? Resources { get; set; }
        public ICollection<TaskInfoDTO>? ChildTasks { get; set; }
    }
}

File: FrontendApp/src/app/models/dtos/task-data.dto.ts
import { TaskStatus } from "../../enums/task-status.enum";
import { ResourceDTO } from "./resource.dto";
import { TaskInfoDTO } from "./task-info.dto";
import { UserDTO } from "./user.dto";

export interface TaskDataDTO {
  id: number;
  companyId: number | null;
  name: string | null;
  description: string | null;
  budget: number | null;
  status: TaskStatus | null;
  completeDate: string | null;
  resources: ResourceDTO[] | null;
  childTasks: TaskInfoDTO[] | null;
  members: UserDTO[] | null;
}
